examples/6-conditional-handlers.php

```diff
@@ -0,0 +1,82 @@
+<?php
+require __DIR__ . '/../vendor/autoload.php';
+require __DIR__ . '/repeated-sample-code.php';
+
+/**
+ * Tactician is meant to be very customizable, it makes no assumptions about
+ * your code.
+ *
+ * For example, let's say you like to handle command X in one way and
+ * command Y in another.
+ *
+ * We can write a custom middleware for that.
+ */
+
+use League\Tactician\Command;
+use League\Tactician\CommandBus;
+use League\Tactician\StandardCommandBus;
+use League\Tactician\Middleware;
+
+// External command
+interface ExternalCommand extends Command
+{
+}
+
+// We create some custom middleware that will forward any ExternalCommand to
+// a separate CommandBus.
+final class ExternalCommandMiddleware implements Middleware
+{
+    /**
+     * @var CommandBus
+     */
+    private $commandBus;
+
+    /**
+     * @param CommandBus $commandBus
+     */
```

Comment:
You can probably drop the docblocks here for brevity. It's pretty clear from the naming and typehints.